azurelinuxagent/ga/update.py

```diff
@@ -1529,26 +1525,12 @@ def __init__(self, path=None, pkg=None, is_fast_track_goal_state=False, host=Non
             self._ensure_downloaded()
             self._ensure_loaded()
         except Exception as e:
-            if isinstance(e, ResourceGoneError):
```

Comment:
ResourceGoneError is already handled in the common download logic, no need to treat it as a special case here.